apps/web/pages/api/twilio/webhook.ts

```diff
@@ -0,0 +1,140 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { CreditService } from "@calcom/features/ee/billing/credit-service";
+import * as twilio from "@calcom/features/ee/workflows/lib/reminders/providers/twilioProvider";
+import { IS_SMS_CREDITS_ENABLED, WEBAPP_URL } from "@calcom/lib/constants";
+import getOrgIdFromMemberOrTeamId from "@calcom/lib/getOrgIdFromMemberOrTeamId";
+import { defaultHandler } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+/*
+  Twilio status callback: creates expense log when sms is delivered or undelivered
+*/
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const signature = req.headers["x-twilio-signature"];
+  const baseUrl = `${WEBAPP_URL}/api/twilio/webhook`;
+
+  const queryParams = new URLSearchParams(req.query as Record<string, string>).toString();
+  const requestUrl = queryParams ? `${baseUrl}?${queryParams}` : baseUrl;
+
+  if (typeof signature === "string") {
+    const isSignatureValid = await twilio.validateWebhookRequest({
+      requestUrl,
+      signature,
+      params: req.body,
+    });
+    if (isSignatureValid) {
+      const messageStatus = req.body.MessageStatus;
+      const { userId, teamId, bookingUid } = req.query;
+
+      if (messageStatus === "delivered" || messageStatus === "undelivered") {
+        if (!IS_SMS_CREDITS_ENABLED) {
+          return res.status(200).send(`SMS credits are not enabled.`);
+        }
+
+        const countryCode = await twilio.getCountryCodeForNumber(req.body.To);
+
+        const smsSid = req.body.SmsSid;
+
+        const parsedTeamId = teamId && !isNaN(Number(teamId)) ? Number(teamId) : undefined;
+        const parsedUserId = userId && !isNaN(Number(userId)) ? Number(userId) : undefined;
+
+        const parsedBookingUid = bookingUid as string;
+
+        if (!parsedUserId && !parsedTeamId) {
+          return res.status(401).send("Team or user id is required");
+        }
+
+        const creditService = new CreditService();
+
+        if (countryCode === "US" || countryCode === "CA") {
+          // SMS to US and CA are free on a team plan
+          let teamIdToCharge = parsedTeamId;
+
+          if (!teamIdToCharge && parsedUserId) {
+            const teamMembership = await prisma.membership.findFirst({
```

Comment:
question: is it okay to use findFirst? what happens if this user belongs to multiple teams?